modules/mod-deep-learning/ModelCardPanel.h

```diff
@@ -0,0 +1,146 @@
+
+/**********************************************************************
+
+   Audacity: A Digital Audio Editor
+   Audacity(R) is copyright (c) 1999-2021 Audacity Team.
+
+   ModelCardPanel.h
+   Hugo Flores Garcia
+
+******************************************************************/
+/**
+
+\file ModelCardPanel.h
+\brief ModelCardPanel implements UI panels to display deep learning model cards.
+
+*/
+/*******************************************************************/
+
+#pragma once
+
+#include "DeepModelManager.h"
+#include "ModelCard.h"
+#include "wx/colour.h"
+
+class ActiveModel;
+class Effect;
+class ShuttleGui;
+
+class ModelCardPanel /* not final */: public wxPanelWrapper
+{
+public:
+   ModelCardPanel(wxWindow *parent, wxWindowID winid, 
+                  ModelCardHolder card, Effect *effect,
+                  std::shared_ptr<ActiveModel> pActiveModel,
+                  const wxSize& size);
+
+   virtual void PopulateOrExchange(ShuttleGui &S) = 0;
+   bool TransferDataToWindow() override;
+   bool TransferDataFromWindow() override;
+
+   // calbacks
+   void OnInstall(wxCommandEvent &event);
+   void OnCancelInstall(wxCommandEvent &event);
+   void OnUninstall(wxCommandEvent &event);
+
+   void OnEnable(wxCommandEvent &event);
+   void OnSelect(wxCommandEvent &event);
+   void OnApply(wxCommandEvent &event);
+   void OnMoreInfo(wxCommandEvent &event);
+
+   void OnClick(wxMouseEvent &event);
+
+   ModelCardHolder GetCard() const { return mCard; }
+
+
+   enum class InstallStatus 
+   {
+      Uninstalled, 
+      Installing, 
+      Installed
+   };
+
+   enum class ModelStatus
+   {
+      Enabled,
+      Disabled
+   };
+
+   void SetInstallStatus(InstallStatus status);
+   void SetModelStatus(ModelStatus status);
+   void PopulateWithNewCard(ModelCardHolder card);
+
+protected:
+   std::map<InstallStatus, wxColour> mInstallStatusColors = {
+      { InstallStatus::Uninstalled, wxColour("#CF6377") },
+      { InstallStatus::Installing,  wxColour(233, 196, 106) },
+      { InstallStatus::Installed,   wxColour(42, 157, 143) }
```

Comment:
I made some trial runs, and the color choices should be adjusted.  The yellow color for "installing" made a poor contrast between text and background which was hard to read.  But this is one of the easiest things to change.

Reply:
1c33a378e460e0e1b8d7cd9028b13f906646f4b3